File: Excepciones/ContadorLetras.java
import java.util.Scanner;

public class ContadorLetras {
    
    public ContadorLetras(String palabra) {
        int[] cuenta = new int[26];

        // convierte la palabra a mayusculas
        palabra = palabra.toUpperCase();
      
        // cuenta la frecuencia de cada letra en la palabra
        for (int i = 0; i < palabra.length(); i++)
            cuenta[palabra.charAt(i) - 'A']++;
      
        // muestra las frecuencias
        System.out.println();
        for (int i = 0; i < cuenta.length; i++)
            if (cuenta[i] != 0)
                System.out.println((char)(i +'A') + ": " + cuenta[i]);
    }
    
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        // palabra el usuario
        System.out.print("Ingresa una palabra (solo letras por favor): ");
        String palabra = scan.nextLine();
        new ContadorLetras(palabra);
    }
}

File: Excepciones/contador_letras.py
class ContadorLetras():

    def __init__(self, palabra):
        cuenta = [0] * 26

        # convierte la palabra a mayusculas
        palabra = palabra.upper()
      
        # cuenta la frecuencia de cada letra en la palabra
        for i in range(len(palabra)):
            cuenta[ord(palabra[i]) - ord('A')] = cuenta[ord(palabra[i]) - ord('A')] + 1
      
        # muestra las frecuencias
        print()
        for i in range(len(cuenta)):
            if cuenta[i] != 0:
                print(chr(i + ord('A')) + ': ' + str(cuenta[i]))


def main():
    palabra = input('Ingresa una palabra (solo letras por favor): ')
    ContadorLetras(palabra)

main()
